discord/http.py

```diff
@@ -973,6 +973,21 @@ def create_invite(self, channel_id, *, reason=None, **options):
             'unique': options.get('unique', True),
         }
 
+        target_type = options.get("target_type")
+
+        if target_type:
+            payload['target_type'] = target_type.value 
+
+        target_user = options.get('target_user')
+
+        if target_user:
+            payload['target_user_id'] = target_user.id
+
+        target_application = options.get('target_application_id')
+
+        if target_application:
+            payload['target_application_id'] = str(target_application)
```

Comment:
Generally speaking, the only data types that I handle within `http.py` are `(str, int, float, bool, None, File)`. Attribute access and enums are done in the higher level methods before being passed down to the lower level http client.